File: LeetCode/cpp/775.cpp
class Solution {
public:
    bool isIdealPermutation(vector<int> &A) {
        int size = A.size(), m = INT_MIN;
        for (int i = 0; i < size - 2; ++i) {
            m = max(m, A[i]);
            if (m > A[i + 2])
                return false;
        }
        return true;
    }
};

File: LeetCode/python3/775.py
class Solution:
    def isIdealPermutation(self, A):
        """
        :type A: List[int]
        :rtype: bool
        """        
        size, m = len(A), 0
        for i in range(size - 2):
            m = max(m, A[i])
            if m > A[i + 2]:
                return False
        return True
